network/containerizer/bridgepolicy.go

```diff
@@ -203,6 +203,27 @@ func (p *BridgePolicy) findSpacesAndDevicesForContainer(
 			guest.Id(), err)
 		return nil, nil, errors.Trace(err)
 	}
+
+	// OVS bridges expose one of the internal ports as a device with the
+	// same name as the bridge. These special interfaces are not detected
+	// as bridge devices but rather appear as regular NICs. If the configured
+	// networking method is "provider", we need to patch the type of these
+	// devices so they appear as bridges to allow the bridge policy logic
+	// to make use of them.
+	if p.containerNetworkingMethod == "provider" {
```

Comment:
I wasn't sure this was the correct thing to do at first, but I read through all the associated logic and it looks like it actually is.

It would be lovely to simplify all this bridge-policy logic; I don't think it needs to be as convoluted as it is.